### PR TITLE
Add verify-go make target

### PR DESCRIPTION
**Pull request by saschagrunert**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
This target can be used to run the latest version of golangci-lint
locally without having the binary installed. This also ensures
reproducible builds due to linting with a constant version.

**Which issue(s) this PR fixes**:

None

**Special notes for your reviewer**:

None

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
> @saschagrunert -- I'd suggest instead just copying in the scripts from https://github.com/kubernetes/repo-infra/tree/master/hack and adapting those to our use.
> /hold

Hm, my main target was to have something not relying on bazel (https://github.com/kubernetes/repo-infra/blob/master/hack/verify-golangci-lint.sh does). Beside this, the script in k/repo-infra is using go modules to install golangci-lint, which is not the recommended method (see https://github.com/golangci/golangci-lint/issues/800#issuecomment-539886088)

**Comment by justaugustus** [comment]
@saschagrunert -- what I meant by "adapting" was removing the bazel-isms.

File: hack/verify-golangci-lint.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+if [[ ! -f .golangci.yml ]]; then
+    echo 'ERROR: missing .golangci.yml in repo root' >&2
+    exit 1
+fi
+
+curl -sfL $URL/$VERSION/install.sh | sh -s $VERSION
+
+$GOLANGCI_LINT version
+$GOLANGCI_LINT linters
+$GOLANGCI_LINT run "$@"
```
Comment (saschagrunert) [inline hack/verify-golangci-lint.sh]: @justaugustus like this :point_up: ?

File: hack/verify-golangci-lint.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+VERSION=v1.23.8
+URL=https://raw.githubusercontent.com/golangci/golangci-lint
```
Comment (justaugustus) [inline hack/verify-golangci-lint.sh]: Use `BASE_URL` or `URL_BASE` and then construct `INSTALL_URL` or `URL` before handing it to curl.

File: hack/verify-golangci-lint.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+GOLANGCI_LINT=./bin/golangci-lint
```
Comment (justaugustus) [inline hack/verify-golangci-lint.sh]:
```suggestion
GOLANGCI_LINT=$(which golangci-lint)
```

We shouldn't make the assumption that the user will have the command in a specific place.

File: hack/verify-golangci-lint.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+if [[ ! -f .golangci.yml ]]; then
+    echo 'ERROR: missing .golangci.yml in repo root' >&2
+    exit 1
+fi
+
+curl -sfL $URL/$VERSION/install.sh | sh -s $VERSION
```
Comment (justaugustus) [inline hack/verify-golangci-lint.sh]: It seems we're unconditionally installing the tool every time we run the script?
We should do something like:
- check tool exists using `which`
- check tool is the correct version
- if tool doesn't exist, then download it
- run the tool

File: hack/verify-golangci-lint.sh (hunk 2 above)
Comment (saschagrunert) [reply hack/verify-golangci-lint.sh]: Fixed :+1:

File: hack/verify-golangci-lint.sh (hunk 4 above)
Comment (saschagrunert) [reply hack/verify-golangci-lint.sh]: I changed the script to:
- download only if `golangci-lint` is not in `$PATH`
- modify path if `golangci-lint` got downloaded

File: hack/verify-golangci-lint.sh (hunk 3 above)
Comment (saschagrunert) [reply hack/verify-golangci-lint.sh]: See https://github.com/kubernetes/release/pull/1153#discussion_r389538584

**Pull request merged**